### PR TITLE
Improve ros bag recording

### PR DESCRIPTION
**Pull request by hect95**
This PR mainly improves rosbag recording for big messages (pointclouds and images) 

- `record_rosbag.sh`
  -  `max-cache-size` is significantly increased to avoid loosing data while writing 
  -  rosbags are splitted every 10 GiB 
  - use `mcap_cfg.yaml` for configuring storage
    - Data is not compressed to avoid overheads when writing (`compression: "None"`)
- More topics are added for recording, this includes dataspeed DBW report topics
- `dev.sh` allows `-p` arg to specity a docker volume where rosbags are going to be saved locally
  - The script will try to use an in-vehicle only directory and will exit if not found
- Set Cyclone DDS as ROS RMW and configure it for high msg throughput
- Foxglove bridge now runs with `num_threads:=0`

### CODE REVIEW

File: dev.sh [hunk 1]
```diff
@@ -3,21 +3,58 @@
...
-
 # Run docker image with local code volumes for development
 docker run -it --rm --net host --privileged \
     -v /dev/shm:/dev/shm \
-    -v $SCRIPT_DIR/rosbags:/opt/ros_ws/rosbags \
+    -v $ROSBAGS_DIR:/opt/ros_ws/rosbags \
     -v $SCRIPT_DIR/scripts/container_tools:/opt/ros_ws/container_tools \
     -v $SCRIPT_DIR/config:/opt/ros_ws/config \
-    av_tools_humble:latest
+    av_tools:latest-dev
```
Comment (GreatAlexander) [inline dev.sh:61]: In general I support having a different tag for dev, but we didn't do this for the rest of repos did we?
Did you start using it here for a reason, did it help any use case?

File: dev.sh [hunk 2]
```diff
@@ -3,21 +3,58 @@
...
+            echo "Unknown option: $1"
+            usage
+            ;;
+    esac
+    shift
+done
+
+# Verify ROSBAGS_DIR exists
+if [ ! -d "$ROSBAGS_DIR" ]; then
+    echo "$ROSBAGS_DIR does not exists! Please provide a valid path to store rosbags"
```
Comment (GreatAlexander) [inline dev.sh]: "does not exist"

File: dev.sh (hunk 1 above)
Comment (hect95) [reply dev.sh:61]: After running `dev.sh` and `runtime.sh` many times in the vehicle's server, I realised I was spending extra time building again the images. So to make the most of docker cache I found it better to use different tags

I did change this in all the repositories

File: dev.sh (hunk 2 above)
Comment (hect95) [reply dev.sh]: Done

**Comment by hect95** [comment]
@GreatAlexander I just added a way to syncronise host date/time with container (forgot to do it earlier)

**Review by GreatAlexander — APPROVED** [review]

**Review by GreatAlexander — APPROVED** [review]

**Pull request merged**